content/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Set up load balancer with MetalLB in HPE GreenLake for Private Cloud Enterprise
+date: 2023-09-01T08:59:11.280Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+disable: false
+tags:
+  - Kubernetes
+  - MetalLB
+  - Load Balancer
+  - HPE GreenLake
+  - HPE GreenLake for Private Cloud Enterprise
+  - HPE GreenLake for Containers
+---
+This blog post describes how to set up the load balancer using [MetalLB](https://metallb.universe.tf/) for a K8s cluster in [HPE GreenLake for Private Cloud Enterprise](https://www.hpe.com/us/en/greenlake/private-cloud-enterprise.html). It allows customers to configure load balancing services for their workloads deployed in the K8s clusters. 
+
+### Overview
+
+In [HPE GreenLake for Private Cloud Enterprise: Containers](https://www.hpe.com/us/en/greenlake/containers.html), customers can create services with the type of *NodePort* for their workloads deployed in K8s clusters using the label *hpecp.hpe.com/hpecp-internal-gateway=true*. The services will be automatically exposed to a container platform gateway host with assigned ports. The deployed workloads will become accessible externally using the gateway host name and the assigned ports as access URLs. Different from various public cloud providers, such as *GCP*, *AWS* and *Microsoft Azure*, HPE GreenLake for Containers doesn’t support network load balancers. When you create services with the type of *LoadBalancer*, they will remain in the *'pending'* state indefinitely when created. This blog post will show you how to use MetalLB to provide load balancing services for K8s clusters in HPE GreenLake for Containers. It provides customers with the flexibility to configure custom load balancers for their deployed workloads in HPE GreenLake for Private Cloud Enterprise.
```

Comment:
Isn't it a problem to clearly say, we don't support network load balancers? Seems like you are exposing a bug when you describe the pending state. I would rephrase in a more positive way.

Also you use HPE GreenLake for Contain twice although you used HPE GreenLake for Private Cloud Enterprise: Containers. As always we are very confusing with our naming convention :-)

Reply:
I'm pretty sure the HPE GreenLake for Containers is a separate entity... I found it on this website: https://www.hpe.com/us/en/greenlake/containers.html

Reply:
> Isn't it a problem to clearly say, we don't support network load balancers? Seems like you are exposing a bug when you describe the pending state. I would rephrase in a more positive way.
> 
> Also you use HPE GreenLake for Contain twice although you used HPE GreenLake for Private Cloud Enterprise: Containers. As always we are very confusing with our naming convention :-)

This is a very good comment. I will rephrase the load balancer support - see below. I am also deleting the pending state in the loadbalancer service. It's a consequence of missing load balancer support, it's not a bug.

"HPE GreenLake for Private Cloud Enterprise doesn’t support network load balancers by default. As cluster administrators, after you create K8s cluster, you can integrate cluster with any load balancers in place to support K8s services of type *LoadBalancer*. "